src/swap-widget/state.tsx

```diff
@@ -470,7 +513,35 @@ export const useTokenBalnces = (tokens: TokenMetadata[], AccountId: string) => {
 
       setBalances(balancesMap);
     });
-  }, [tokens.map(t => t?.id).join('-')]);
+  }, [tokens, AccountId]);
+
+  useEffect(() => {
+    initTokenBalances();
+    const interval = setInterval(initTokenBalances, 60000);
+    return () => clearInterval(interval);
+  }, [initTokenBalances]);
+
+  const updateTokenBalance = (tokenIds: string[], AccountId: string) => {
+    tokenIds.forEach(tokenId => {
+      ftGetBalance(tokenId, AccountId).then(available => {
+        setBalances({
+          ...balances,
+          [tokenId]: toReadableNumber(
+            tokens.find(t => t.id === tokenId)?.decimals || 0,
+            available
+          ),
+        });
+      });
+    });
+  };
+
+  const tokenDeps = tokens.map(t => t?.id).join('-');
+  useEffect(() => initTokenBalances(), [
```

Comment:
you can combine interval and initializations in one hook to avoid double initTokenBalances call.

Plus you can declare initTokenBalances function just inside the hook